orbit/pkg/update/escrow_buddy.go

```diff
@@ -0,0 +1,120 @@
+package update
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/fleetdm/fleet/v4/server/fleet"
+	"github.com/rs/zerolog/log"
+)
+
+// EscrowBuddyRunner sets up [Escrow Buddy][1] to rotate FileVault keys on
+// macOS without user interaction. This runner:
+//
+// - Ensures Escrow Buddy is added as a target for the update runner, so the
+// authorization plugin is downloaded and installed.
+// - Shells out to call `defaults` to configure Escrow Buddy according to
+// server instructions provided via notifications.
+//
+// [1]: https://github.com/macadmins/escrow-buddy
+type EscrowBuddyRunner struct {
+	// updateRunner is the wrapped Runner where Escrow Buddy will be set as
+	// a target.
+	updateRunner *Runner
+	// runCmdFunc can be set in tests to mock the command executed to
+	// configure Escrow Buddy
+	runCmdFunc func(cmd string, args ...string) error
+	// runMu guards runs to prevent multiple Run calls happening at the
+	// same time.
+	runMu sync.Mutex
+	// lastRun is used to guarantee that the run interval is enforced
+	lastRun time.Time
+	// interval defines how often Run is allowed to perform work
+	interval time.Duration
+}
+
+// NewEscrowBuddyRunner returns a new instance configured with the provided values
+func NewEscrowBuddyRunner(runner *Runner, interval time.Duration) fleet.OrbitConfigReceiver {
+	return &EscrowBuddyRunner{updateRunner: runner, interval: interval}
+}
+
+func (e *EscrowBuddyRunner) Run(cfg *fleet.OrbitConfig) error {
+	log.Debug().Msgf("EscrowBuddyRunner: notification: %t", cfg.Notifications.RotateDiskEncryptionKey)
+
+	if e.updateRunner == nil {
+		log.Debug().Msg("EscrowBuddyRunner: received nil UpdateRunner, this probably indicates that updates are turned off. Skipping any actions related to Disk encryption")
```

Comment:
Just double-checking if that should be promoted to an `Info` log to better understand if/why rotation doesn't happen? Maybe it would get too noisy though...